src/__tests__/layouts/Layout.test.tsx

```diff
@@ -1,13 +1,35 @@
-import renderWithMockedProvider from "../utils";
 import Layout from "../../layouts";
-import Home from "../../pages";
+import { screen, render } from "@testing-library/react";
+import Custom404 from "@/pages/404";
+import { RouterContext } from "next/dist/shared/lib/router-context";
+import { createMockRouter } from "../../test-utils/createMockRouter";
 
 describe("Header", () => {
-  it("shoud render layout success", () => {
-    renderWithMockedProvider(
-      <Layout>
-        <Home />
-      </Layout>
+  it("should render the navBar and 404", () => {
+    render(
+      <RouterContext.Provider value={createMockRouter({ pathname: "/404" })}>
+        <Layout>
+          <Custom404 />
+        </Layout>
+      </RouterContext.Provider>
```

Comment:
Persoanally, I think this description is not correct. I think it should focus on the layout itself. Also, I think  custom404 should not be here becasue we are writing unit test for "Layout"